bin/propolis-server/src/lib/migrate/destination.rs

```diff
@@ -86,12 +87,14 @@ impl DestinationProtocol {
                 Err(MigrateError::UnexpectedMessage)
             }
         }?;
-        info!(self.log(), "Src read Preamble: {:?}", preamble);
-        // XXX: For demonstration purposes only.
-        if preamble.vm_descr.vcpus != vec![0u32, 1, 2, 3] {
+        info!(self.log(), "Destination read Preamble: {:?}", preamble);
+        if !preamble
+            .instance_spec
+            .is_migration_compatible(&self.mctx.instance_spec)
+        {
             error!(
                 self.log(),
-                "invalid CPU count in preamble ({:?})", preamble.vm_descr.vcpus
+                "Source and destination instance specs incompatible"
```

Comment:
Should we dump both instance specs here to help diagnose this? 
Not blocking on this PR but perhaps it might make sense in the future to adjust `is_migration_compatible` to return a `Result<(), ErrOfSomeKind>` instead of just a `bool`.

Reply:
Yeah, that's a good call. Filed #178 to track.